snippets/facets.liquid

```diff
@@ -381,6 +381,7 @@
   {%- endif -%}
   {% comment %}  Drawer and mobile filter {% endcomment %}
   <menu-drawer class="mobile-facets__wrapper{% if filter_type == 'horizontal' or filter_type == 'vertical' %} medium-hide large-up-hide{% endif %}" data-breakpoint="mobile">
+    <h2 class="visually-hidden" id="verticalTitle">{{ 'products.facets.filter_by_label' | t }}</h2>
```

Comment:
@svinkle 👀

Reply:
Another option is what @svinkle  suggested in the original issue and simplifying to use aria-label with a string value. 

https://github.com/Shopify/dawn/pull/1443#discussion_r838874860

Reply:
I believe we can have duplicate aria-labels since it's just text no? It wouldn't be connected to an id

Reply:
Reading Scott's recommendations, my understanding is that the `aria-label` would replace `aria-labelledby` on the `<aside>` element. We would not be able to have two `aria-label`s on the same `aside` element.

Reply:
Okay, so I've tested this a bit and this is my proposal. 
```
<aside>
  <span id="verticalTitle" class="visually-hidden" aria-hidden="true">
    <span class="small-hide">Desktop specific heading</span>
    <span class="medium-hide large-up-hide">Mobile specific heading</span>
  </span>
</aside>
```

I propose that we introduce a new visually hidden element that has the `verticalTitle` id. This will be used to contain the mobile _and_ desktop headings. 

Why? 
* Using a single element that contains the mobile and heading text means that we can avoid the duplicate ID problem. It introduces duplicate logic for the title within the same file, but I think the benefits outweigh the downsides of introducing duplicate code in this instance.
* Using classes we can show (`display: block`) and hide (`display: none`) the text based on the viewport. Meaning, the correct text is shared with the screenreader in the right context. 
  * I validated this with the chrome Accessibility tree. The `<aside>` correctly read the text based on the viewport. 
  * Using VO + Safari, it read `Desktop specific heading, Complimentary` on Desktop and `Mobile specific heading, Complimentary` when navigating to the `<aside>` using the virtual cursor 
  * Using VO + Safari, I was able to use the rotor to view the landmarks on the page. It correct showed `Desktop specific heading` and `Mobile specific heading` in the right context/breakpoint. 
* Including this in one location outside of the drawer, but within the `aside` means that this will apply for all filtering/sort contexts (vertical, drawer and horizontal, enabled=true/false)
  
Additional considerations 
* Using `aria-hidden="true"`, we can remove this content from the accessibility tree. This means that we can use the element to read the correct information for the `<aside>` element. At the same time, this content is hidden from users who are using the virtual cursor and removes duplicate/redundant content. 
* Nested `<span>`s is considered valid HTML. Open to another option for the parent element.

@svinkle Would you be okay with this proposal? 
I haven't had time to test this with any other screenreader than VO. So I cannot confirm that it would work in every instance, at this moment.

Reply:
Could we use the drawer/horizontal/vertical setting as a conditional?

If we can't distinguish for specific strings, not worries. "Filter" will suffice.